web/src/message_fetch.ts

```diff
@@ -340,7 +340,25 @@ export function load_messages(opts: MessageFetchOptions, attempt = 1): void {
         if (page_params.narrow !== undefined) {
             terms = [...terms, ...page_params.narrow];
         }
-        data.narrow = JSON.stringify(terms);
+        // TODO(stream_id): The server would ideally work with stream ids instead
+        // of stream names.
+        const server_terms = [];
+        for (const term of terms) {
+            if (term.operator === "channel") {
+                const sub = stream_data.get_sub_by_id_string(term.operand);
+                server_terms.push({
+                    ...term,
+                    // If the sub is undefined, we'll get an error which is handled
+                    // later by showing a "this channel does not exist or is private"
+                    // notice.
+                    operand: sub?.name,
```

Comment:
This caused a bug (https://chat.zulip.org/#narrow/stream/464-kandra-js-errors/topic/ZodError.3A.20.5B)

Reply:
#31728